storage/include/sparse_merkle/base_types.h

```diff
@@ -68,25 +69,56 @@ class Hash {
   static constexpr const char* const HASH_ALGORITHM = "SHA3-256";
 
   static constexpr size_t SIZE_IN_BYTES = SIZE_IN_BITS / 8;
-  static constexpr size_t MAX_NIBBLES = SIZE_IN_BITS / 16;
+  static constexpr size_t MAX_NIBBLES = SIZE_IN_BYTES * 2;
 
   // This is the hash of the empty string ''. It's used as a placeholder value
   // for merkle tree hash calculations.
   //
   // Change this if the hash algorithm changes
-  static constexpr std::array<uint8_t, SIZE_IN_BYTES> EMPTY = {
+  static constexpr std::array<uint8_t, SIZE_IN_BYTES> EMPTY_BUF = {
       0xa7, 0xff, 0xc6, 0xf8, 0xbf, 0x1e, 0xd7, 0x66, 0x51, 0xc1, 0x47, 0x56, 0xa0, 0x61, 0xd6, 0x62,
       0xf5, 0x80, 0xff, 0x4d, 0xe4, 0x3b, 0x49, 0xfa, 0x82, 0xd8, 0x0a, 0x4b, 0x80, 0xf8, 0x43, 0x4a};
 
+  Hash() {}
+
+  Hash(std::array<uint8_t, SIZE_IN_BYTES> buf) : buf_(buf) {}
+
+  const uint8_t* data() const { return buf_.data(); }
+  size_t size() const { return buf_.size(); }
+
```

Comment:
It's strange to see classes defined in the file called base_types.h :-)

Reply:
I agree. I just couldn't think of a better name. I'm happy to call it something else. They are the basic types needed to build a merkle tree other than the nodes of the trees and keys themselves.